librosa/sequence.py

```diff
@@ -359,6 +361,341 @@ def __dtw_backtracking(D_steps, step_sizes_sigma):  # pragma: no cover
+
+            backtrack[i, j] = np.argmax(vec[:init_limit])
+            score[i, j] = max(0, vec[backtrack[i, j]])
+            # Is it a reset?
+            if score[i, j] == 0:
+                backtrack[i, j] = -1
+
+    # Now fill in the rest of the table
+    for i in range(2, sim.shape[0]):
+        for j in range(2, sim.shape[1]):
+            score_values[:] = (score[i-1, j-1], score[i-1, j-2], score[i-2, j-1])
+            sim_values[:] = (sim[i-1, j-1], sim[i-1, j-2], sim[i-2, j-1])
+            t_values = sim_values > 0
+            if sim[i, j] > 0:
+                # if knight is true, it's max of (-1,-1), (-1, -2), (-2, -1)
+                # otherwise, it's just the diagonal move (-1, -1)
+                # for backtracking purposes, if the max is 0 then it's the start of a new sequence
+                # if the max is non-zero, then we extend the existing sequence
+                backtrack[i, j] = np.argmax(score_values[:limit])
+                score[i, j] = score_values[backtrack[i, j]] + sim[i, j]  # or + 1 for binary
+
+            else:
+                # if the max of our options is negative, then it's a hard reset
+                # otherwise, it's a skip move
+                vec[:limit] = (score_values[:limit]
+                               - t_values[:limit] * gap_onset
+                               - (~t_values[:limit]) * gap_extend)
+
+                backtrack[i, j] = np.argmax(vec[:limit])
+                score[i, j] = max(0, vec[backtrack[i, j]])
+                # Is it a reset?
+                if score[i, j] == 0:
+                    backtrack[i, j] = -1
+
+    return score, backtrack
+
+
+def __rqa_backtrack(score, pointers):
+    '''RQA path backtracking
+
+    Given the score matrix and backtracking index array,
+    reconstruct the optimal path.
+    '''
+
+    # backtracking rubric:
+    #   0 ==> diagonal move
+    #   1 ==> knight move up
+    #   2 ==> knight move left
+    #  -1 ==> reset (sim = 0)
+    #  -2 ==> start of sequence (sim > 0)
+
+    # This array maps the backtracking values to the
+    # relative index offsets
+    offsets = [(-1, -1), (-1, -2), (-2, -1)]
+
+    # Find the maximum to end the path
+    idx = list(np.unravel_index(np.argmax(score), score.shape))
+
+    # Construct the path
+    path = []
+    while True:
+        bt_index = pointers[tuple(idx)]
+
+        # A -1 indicates a non-inclusive reset
+        # this can only happen when sim[idx] == 0,
+        # and a reset with zero score should not be included
+        # in the path.  In this case, we're done.
+        if bt_index == -1:
+            break
+
+        # Other bt_index values are okay for inclusion
+        path.insert(0, idx)
+
+        # -2 indicates beginning of sequence,
+        # so we can't backtrack any further
+        if bt_index == -2:
+            break
+
+        # Otherwise, prepend this index and continue
+        idx = [idx[_] + offsets[bt_index][_] for _ in range(len(idx))]
```

Comment:
is there a reason why you're iterating with the underscore symbol rather than with a letter?

Reply:
code checker does not like single-letter variables; `_` for this purpose is pythonic though.